File: solved.ac/python/14405.py

s = input()
canDo = ["pi", "ka", "chu"]
for i in range(3):
  s = s.replace(canDo[i], "*")
for i in range(len(s)):
  if ord('a') <= ord(s[i]) <= ord('z'):
    print("NO")
    exit()
print("YES")
# for i in range(3):
#   temp = s.split(canDo[i])
#   s = ''.join(temp)
#   if not len(s) : 
#     print("YES")
#     sys.exit()
# print("NO")

File: solved.ac/javascript/14405.js
const fs = require("fs")
let input = fs.readFileSync("./dev/stdin").toString().trim()
temp = ["pi", "ka", "chu"]
for (let i = 0; i < 3; i++) {
  input = input.replaceAll(temp[i], "*")
}
let flag = true
for (let i = 0; i < input.length; i++) {
  // console.log(input[i].charCodeAt(), "a".charCodeAt())
  if (
    input[i].charCodeAt() >= "a".charCodeAt() &&
    input[i].charCodeAt() <= "z".charCodeAt()
  ) {
    console.log("NO")
    flag = false
    break
  }
}
if (flag) {
  console.log("YES")
}
